python/lsst/ctrl/execute/slurmPlugin.py

```diff
@@ -162,3 +170,135 @@ def createAllocationFile(self, input):
             print("Wrote new Slurm job allocation bash script to %s" % outfile)
         os.chmod(outfile, 0o755)
         return outfile
+
+    def glideinsFromJobPressure(self):
+        """Calculate the number of glideins needed from job pressure
+
+        Returns
+        -------
+        number : `str`
+            The number of glideins
+        """
+
+        import math
+        import socket
+
+        import htcondor
+        from lsst.ctrl.bps.htcondor import condor_q
+
+        verbose = self.isVerbose()
+
+        maxNumberOfGlideins = self.getNodes()
+        coresPerGlidein = self.getCPUs()
+        ratioMemCore = self.getMemoryPerCore()
+        auser = self.getUserName()
+
+        # initialize counters
+        totalCores = 0
+
+        try:
+            schedd_name = socket.getfqdn()
+            coll = htcondor.Collector()
+            schedd_ad = coll.locate(htcondor.DaemonTypes.Schedd)
+            scheddref = htcondor.Schedd(schedd_ad)
+            projection = [
+                "JobStatus",
+                "Owner",
+                "RequestCpus",
+                "JobUniverse",
+                "RequestMemory",
+            ]
```

Comment:
I added comments on the classads to be returned, and the constraint.

Reply:
and it can be emphasized that there are classads to be returned (the first list), and then the constraint (a second list.)   All of the classads there do not have to be returned but they seemed useful to have available for debugging.